state/protocol/badger/snapshot_test.go

```diff
@@ -301,12 +301,78 @@ func TestSealingSegment(t *testing.T) {
 			require.NoError(t, err)
 
 			// sealing segment should be [B2, B3, B4]
-			assert.Len(t, segment, 3)
-			assert.Equal(t, block2.ID(), segment[0].ID())
-			assert.Equal(t, block3.ID(), segment[1].ID())
-			assert.Equal(t, block4.ID(), segment[2].ID())
+			assert.Len(t, segment.Blocks, 3)
+			assert.Equal(t, block2.ID(), segment.Blocks[0].ID())
+			assert.Equal(t, block3.ID(), segment.Blocks[1].ID())
+			assert.Equal(t, block4.ID(), segment.Blocks[2].ID())
+		})
+	})
+
+	// test sealing segment where a block contains a ExecutionReceipt that
+	// references a ExecutionResult contained in a different block.
+	// the sealing segment should contain any ExecutionResults that were
+	// missing from a blocks payload in the SealingSegment.ExecutionResults field
+	// ROOT -> B1(Rec_1) -> B2[Res_1, Rec_2, Res_2] -> B3(Rec_3, Res_3, Seal_B1) -> B4 -> B5(Rec_4, Res_4, Seal_B4)
```

Comment:
` B5(Rec_4, Res_4, Seal_B4)`

This is not a valid case, because, seals can't be included in the same block as the result. So technically , the sealingSegment length has to be 3, because the result/receipt can't be in the same block as the executed block, and the seal can't be in the same block as the result/receipt. 

I'd like to request the tests for 2 other cases:
1. 
`A <- B[Result_A, Receipt_A_1] <- C[Result_B, Receipt_B, Receipt_A_2] <- D <- E <- F (Seal_C)`

The sealing segment for block F is `[C,D,E,F] + [result_A]`, `C` has `Receipt_A_2`, whose result is in block B

2.
`A <- B[Result_A, Receipt_A_1] <- C[Result_B, Receipt_B, Receipt_A_2] <- D (Receipt_B_2, Receipt_A_3) <- E <- F (Seal_C)`

The sealing segment for block F is `[C,D,E,F] + [resultA]`, because even though block C has a `receiptA_2`, and block D has a `receipt_A_3`, both receipts have the same result, so resultA should only be added once.